core/Common.php

```diff
@@ -553,10 +553,12 @@ public static function getRequestVar($varName, $varDefault = null, $varType = nu
                     $ok = true;
                 }
             } elseif ($varType === 'float') {
-                $valueToCompare = (string)(float)$value;
-                $valueToCompare = Common::forceDotAsSeparatorForDecimalPoint($valueToCompare);
+                $valueToCompare = Common::forceDotAsSeparatorForDecimalPoint($value);
 
-                if ($value == $valueToCompare) {
+                // simplyfied regex for float; without support for underscore notation
+                $floatRegex = "/^[+-]?((([0-9]+)|(([0-9]+)?\.([0-9]+))|(([0-9]+)\.([0-9]+)?))([eE][+-]?([0-9]+))?)$/";
```

Comment:
In a comment, we can add at least one example of a value that matches that regex. That would be helpful for others to understand what the regex does.